### ISSUE TITLE
Document backend API

### ISSUE THREAD

**Issue by Half-Shot** [matrix-org/matrix-bifrost#61]
This means things like the interfaces used for backends, when things are generally called and what events should be fired when.

**Comment by ShubhamKarala** [comment]
I want to work on this. Is there any blueprint i should follow?

**Comment by ShubhamKarala** [comment]
I want to work on this!! Kindly guide me from where to start, I have a synapse homeserver running at https://serverless.ooo

**Comment by Half-Shot** [comment]
Hiya! Yeah it can be, though you'll probably need to ask me fairly often what each bit of the code does (which is a good opportunity to learn :) ). 

I guess one thing we really need is the ability to host some documentation for the project online, that can be generated from the typescript code. Is this something you can investigate?

**Comment by ShubhamKarala** [comment]
Yes, we can use typedoc (https://github.com/TypeStrong/typedoc) to generate HTML documentation from typescript code
